x/incentives/keeper/proposals.go

```diff
@@ -0,0 +1,117 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/tharsis/evmos/x/incentives/types"
+)
+
+// RegisterIncentive creates an incentive for a contract
+func (k Keeper) RegisterIncentive(
+	ctx sdk.Context,
+	contract common.Address,
+	allocations sdk.DecCoins,
+	epochs uint32,
+) (*types.Incentive, error) {
+
+	// check if the Incentives are globally enabled
+	params := k.GetParams(ctx)
+	if !params.EnableIncentives {
+		return nil, sdkerrors.Wrap(
+			types.ErrInternalIncentive,
+			"incentives are currently disabled by governance",
+		)
+	}
+
+	// check if the incentive is already registered
+	if k.IsIncentiveRegistered(ctx, contract) {
+		return nil, sdkerrors.Wrapf(
+			types.ErrInternalIncentive,
+			"incentive already registered: %s", contract,
+		)
+	}
+
```

Comment:
@fedekunze I used your recommendation. The resulting MintDemon is `stake`. Is that correct?

Reply:
technically yes because that's the default denom on the SDK